test/api/requests/build.test.js

```diff
@@ -146,6 +145,102 @@ describe("Build API", () => {
     })
   })
 
+  describe("POST /build/:id/restart", () => {
+    context("On the default branch", () => {
+      it("should create a build that mirrors the original", done => {
+        let build
+        
+        factory(Build, { branch: undefined }).then(model => {
+          build = model
+          return User.findOne(build.user)
+        }).then(user => {
+          return session(user)
+        }).then(cookie => {
+          return request("http://localhost:1337")
+            .post(`/v0/build/${build.id}/restart`)
+            .set("Cookie", cookie)
+            .expect(200)
+        }).then(response => {
+          return Build.find({
+            user: build.user,
+            site: build.site,
+            branch: build.branch,
+          })
+        }).then(builds => {
+          expect(builds).to.have.length(2)
+          done()
+        })
+      })
+    })
+
+    context("On a preview branch", () => {
+      it("should create a build that mirrors the original", done => {
+        let build
+        
+        factory(Build, { branch: "preview" }).then(model => {
+          build = model
+          return User.findOne(build.user)
+        }).then(user => {
+          return session(user)
+        }).then(cookie => {
+          return request("http://localhost:1337")
+            .post(`/v0/build/${build.id}/restart`)
+            .set("Cookie", cookie)
+            .expect(200)
+        }).then(response => {
+          return Build.find({
+            user: build.user,
+            site: build.site,
+            branch: build.branch,
+          })
+        }).then(builds => {
+          expect(builds).to.have.length(2)
+          done()
+        })
+      })
+    })
+
+    it("should render a JSON representation of the new build", done => {
+      let build, response
+      
+      factory(Build).then(model => {
+        build = model
+        return User.findOne(build.user)
+      }).then(user => {
+        return session(user)
+      }).then(cookie => {
+        return request("http://localhost:1337")
+          .post(`/v0/build/${build.id}/restart`)
+          .set("Cookie", cookie)
+          .expect(200)
+      }).then(res => {
+        response = res
+
+        return Build.find({
+          user: build.user,
+          site: build.site,
+        }).sort("createdAt DESC")
+      }).then(builds => {
+        buildResponseExpectations(response.body, builds[0])
+        done()
+      })
+    })
+
+    it("should respond with a 403 if the original build is not associated with the user", done => {
```

Comment:
Amended the commit that adds the API endpoint so that now it has this behavior.

Good catch